_posts/2025-03-20-phased-strimzi-upgrade-example.md

```diff
@@ -0,0 +1,637 @@
+- Roll back is an _all or nothing_ operation. If you encounter an issue you would need to pause _all_ Kafka resources again, downgrade the Strimzi Operator and then un-pause the resources.   
+
+### Resource Selector
+
+While the options above work in theory, in practice they have several drawbacks which make them unsuitable for production deployments. 
+A better option comes by using the _Custom Resource Selector_. 
+This is an [environment variable](https://strimzi.io/docs/operators/latest/deploying#ref-operator-cluster-str) that is set in the Cluster Operator deployment:
+
+>`STRIMZI_CUSTOM_RESOURCE_SELECTOR` 
+   Optional. The label selector to filter the custom resources handled by the Cluster Operator. The operator will operate only on those custom resources that have the specified labels set. Resources without these labels will not be seen by the operator. The label selector applies to `Kafka`, `KafkaConnect`, `KafkaBridge`, `KafkaMirrorMaker`, and `KafkaMirrorMaker2` resources. `KafkaRebalance` and `KafkaConnector` resources are operated only when their corresponding Kafka and Kafka Connect clusters have the matching labels.  
+> ```yaml 
+> env:   
+> 	- name: STRIMZI_CUSTOM_RESOURCE_SELECTOR     
+> 	  value: label1=value1,label2=value2 
+> ```
+
+Using this selector you can set a label, `Strimzi-043` for example, in a cluster scoped Strimzi Deployment and then tag all the Kafka CRs (and other operands) with this label and that Strimzi deployment will manage them and only them. 
+Any operand CRs without the label will be ignored. 
+
+You could then deploy _another_ cluster-scoped Strimzi operator with a different custom resource selector, `Strimzi-043-b` for example, and tag a different set of Operand CRs with that label.
+The two Strimzi deployments would coexist happily (provided you have setup independent RBAC -- more on this later), managing completely separate sets of CRs.
+
+This features allows us to create a more robust upgrade process for the Strimzi Operator and its operands.
+
+## Resource Selector based upgrade
+
+Using the Custom Resource Selector (CRS) allows us to have fine grained control over which operands are upgraded to new Strimzi versions and when. 
+However, in order to use this process we need to deploy multiple versions of Strimzi concurrently.
+
+### Concurrent Strimzi Deployments
+
+The standard Strimzi deployment includes service accounts and RBAC resources (`Role`s, `ClusterRole`s and their associated bindings) needed for Strimzi to do its job. 
+By default these are all named `strimzi-cluster-operator` or similar.
+If we try and deploy multiple operator instances in the same namespace, or over multiple namespaces but needing to share `ClusterRole`s and bindings, things will get messy.
+Also, different Strimzi versions may need different permissions and so we need to make sure independent copies of these RBAC resources are deployed for each Strimzi version.
+
+#### Deploying Strimzi with Kustomize
```

Comment:
If you are happy with this, no need to change. But I wonder if it would be easier to just use different namespaces for each operator version?